packages/teleport/src/lib/tdp/codec.ts

```diff
@@ -682,6 +693,37 @@ export default class Codec {
     };
   }
 
+  // | message type (19) | completion_id uint32 | directory_id uint32 | path_length uint32 | path []byte | offset uint64 | length uint32 |
```

Comment:
[out of scope, discussion]
If we will add more request types I think we should create some simple utility:
```ts
const sharedDirectoryReadRequestSchema = schema({
   completion_id: uint32,
   directory_id: uint32,
   path_length: uint32,
   path: byteArray,
   offset: uint64,
   length: uint32
});

type SharedDirectoryReadRequest = ReturnType<typeof sharedDirectoryReadRequestSchema.try_parse>;

const data = sharedDirectoryReadRequestSchema.try_parse(buffer);
```

Reply:
I'm not sure if I'm understanding correctly. `createSchema` is returning a class which generates this `decodeSharedDirectoryReadRequest` function (as `tryParse`)?

Reply:
something like https://codesandbox.io/s/goofy-hermann-vdrx01?file=/src/index.ts